src/cplscheme/MultiCouplingScheme.cpp

```diff
@@ -122,73 +88,58 @@ void MultiCouplingScheme::initializeData()
     }
     sendData();
   }
-
-  // in order to check in advance if initializeData has been called (if necessary)
-  setHasToSendInitData(false);
-  setHasToReceiveInitData(false);
 }
 
-void MultiCouplingScheme::advance()
+void MultiCouplingScheme::explicitAdvance()
 {
-  PRECICE_TRACE(getTimeWindows(), getTime());
-  checkCompletenessRequiredActions();
-
-  PRECICE_CHECK(!hasToReceiveInitData() && !hasToSendInitData(),
-                "initializeData() needs to be called before advance if data has to be initialized!");
+  PRECICE_ASSERT(isImplicitCouplingScheme(), "MultiCouplingScheme is always Implicit.");
+  // TODO this class hierarchy has a smell! I think that MultiCouplingScheme should not be derived from BaseCouplingScheme, but directly from CouplingScheme or an intermediate layer between CouplingScheme and BaseCouplingScheme.
```

Comment:
```suggestion
// @todo implement
```
Nope. We could simply implement the explicit variant of the `MultiCouplingScheme`. The assertion is the right way to go for the moment.